### PR TITLE
Test cases for JENKINS-22566

### PR DESCRIPTION
**Pull request by basil**
See [JENKINS-22566](https://issues.jenkins-ci.org/browse/JENKINS-22566). When a build has a parameter whose value contains an invalid XML character, the XML API returns a 500 Internal Server error.

This change adds test coverage for the case where the parameter value does not contain an invalid XML character. This test passes.

This change also adds test coverage for the case where the parameter value _does_ contain an invalid XML character. This test currently fails due to JENKINS-22566, so the test is ignored. When JENKINS-22566 is fixed, the `@Ignore` annotation should be removed.

### Changelog entry

* N/A

### CODE REVIEW

**Review by oleg-nenashev — APPROVED** [review]
It would be gread to have more descriptive method names, but the rest looks good to me. Thanks!

**Review by MRamonLeon — CHANGES_REQUESTED** [review]
Do you agree to change these method names as @oleg-nenashev suggested? It helps us to better understand the purpose of each test. I've done a proposal but feel free to rename to whatever you consider more appropriated. Thank you @basil.

**Comment by fcojfernandez** [comment]
I agree with @MRamonLeon 's suggestion. Maybe simpler names but more descriptible

**Comment by oleg-nenashev** [comment]
On hold taking the discussion

**Comment by oleg-nenashev** [comment]
So the ATH test will legitimately fail after the merge. It is going to be a chicken-and-egg, because we need to fix the test in ATH which depends on the core itself. So the only proper approach would be to have the ATH logic to behave differently depending on the version.

screenshot 2019-02-01 at 13 54 34

@raul-arabaolaza @olivergondza @MRamonLeon  what would be your take on that?

**Comment by oleg-nenashev** [comment]
Sorry, wrong PR

**Comment by oleg-nenashev** [comment]
@basil it would be great if you accept proposed changes from the WebUI then (new GitHub feature)

**Comment by basil** [comment]
I'm not so sure about the proposed changes above. Here are the existing method names in this file:

- `xpath`
- `json`
- `wrappedZeroItems`
- `xPathDocumentFunction`
- `wrappedOneItem`
- `wrappedMultipleItems`
- `unwrappedZeroItems`
- `unwrappedOneItem`
- `unwrappedLongString`
- `unwrappedMultipleItem`

My new names follow the same convention. The suggestions above do not follow the same convention and lead to inconsistency within the same file. For example, the suggestions above add "test" to the end of each name when none of the existing methods have it.

That having been said, I don't see much value to debating names for their own sake (as opposed to debating the merits of a particular set of _semantics_), so if you would like to rename what I wrote, feel free.

**Comment by oleg-nenashev** [comment]
I agree with @basil 's comment. To save time, let's do a bulk refactoring as a follow-up then

**Review by MRamonLeon — APPROVED** [review]
Ok then

**Pull request merged**
